lib/Alcaeus/MongoDbAdapter/TypeConverter.php

```
@@ -45,6 +45,8 @@ public static function fromLegacy($value)
                 return $value;
             case $value instanceof \DateTimeInterface:
                 return self::fromLegacy((array) $value);
+            case $value instanceof \BackedEnum:
+                return $value->value;
```

All `BackedEnum`s have a `value` property

---

I would handle `instanceof \Enum` as well by using `$value->name`

---

That would deviate from the "pure enums cannot be serialized" MongoDB BSON standard they say in their API.
https://www.mongodb.com/docs/php-library/v1.15/tutorial/modeling-bson-data/#working-with-enums

Which I imagine would make the later conversion to the modern system difficult, do you think it's better to conform to their standard?

---

Ok - can you throw an exception if we try?  Otherwise, they will get serialized, but not how we want them to be.

---

Added. Worth noting that when you `json_encode` a pure enum it just fails silently (returns `false`), which may also be the behavior in the modern library.